src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs

```diff
@@ -115,6 +115,34 @@ internal static int GetTokenLength(string input, int startIndex)
             return input.Length - startIndex;
         }
 
+        [Pure]
+        internal static bool IsToken(string input)
+        {
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (!IsTokenChar(input[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
Oh, I just realized it's `ReadOnlySpan<byte>`, not `ReadOnlySpan<char>`... nevermind.